src/bin/pg_autoctl/pgsql.c

```diff
@@ -873,17 +873,21 @@ pgAutoCtlDebugNoticeProcessor(void *arg, const char *message)
 bool
 pgsql_begin(PGSQL *pgsql)
 {
-	PGconn *connection;
-
-	pgsql->connectionStatementType = PGSQL_CONNECTION_MULTI_STATEMENT;
-	connection = pgsql_open_connection(pgsql);
-	if (connection == NULL)
+	if (!pgsql_execute(pgsql, "BEGIN"))
```

Comment:
pgsql_execute already opens a connection if needed, so no need to do so manually.